A3-channel-tracing.md

```diff
@@ -0,0 +1,204 @@
+Channel Tracing
+----
+* Author(s): Noah Eisen (ncteisen@google.com)
+* Approver: markdroth, ctiller
+* Status: Draft
+* Implemented in: N/A
+* Last updated: 03/09/17
+* Discussion at: https://groups.google.com/forum/#!topic/grpc-io/WFDj3KeHYTI
+
+## Table of Contents
+
+  * [Abstract](#abstract)
+  * [Background](#background)
+  * [Proposal](#proposal)
+  * [Rationale](#rationale)
+  * [Implementation](#implementation)
+     * [Basic Design](#basic-design)
+     * [Tracer Functions](#tracer-functions)
+     * [API Changes](#api-changes)
+     * [Format of Exported Data](#format-of-exported-data)
+     * [Garbage Collection](#garbage-collection)
+
+## Abstract
+
+The document proposes adding a dedicated, in-memory tracing object per channel and subchannel. The tracing object will record important steps in the life of a channel, like address resolution, subchannel creation, channel state changes, etc etc. This trace will be made available with a new gRPC API function call. The long term goal is to incorporate the trace into grpcz or a custom channel tracing UI.
```

Comment:
General comment: Can we be a little more precise with the terminology used in this doc?  In particular, it's not clear to me whether the term "trace" refers to a single trace node or to the collection of all trace nodes in a given tracer.  Let's define the terms and make sure that we're using them consistently throughout the document (and the code).